kong/core/router.lua

```diff
@@ -659,16 +660,12 @@ function _M.new(apis)
       return nil
     end
 
-    local new_uri
-    local uri_root = uri == "/"
+    local uri_root = request_uri == "/"
```

Comment:
some of this logic feels ever-so-slightly hand wavey. some comments here about what's going on would be useful, particularly given the subtlety and complexity of the bug resolved here.